core/modules/admin_bar/css/admin_bar.css

```diff
@@ -78,6 +78,11 @@
   filter: progid:DXImageTransform.Microsoft.gradient(startColorstr='#9F9F9F', endColorstr='#808080',GradientType=0); /* IE6-9 */
 }
 
+#admin-bar li {
```

Comment:
What he said ^

My base styles added bottom margin to all li's and it was making the admin bar taller than it wanted to be.